### PR TITLE
Add warning to CheckIlluminaDir if we detect cycles without data.

### PR DESCRIPTION
**Pull request by jacarey**
CBCLReader will ignore and log a warning for tiles without data.

This fixes an issue where you have a tile definition in the cbcl but the tile has no data to decompress.
In addition we added some warning messages when we encounter a tile with no data.
----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [x] Edited the README / documentation (if applicable)
- [x] All tests passing on Travis

#### Review
- [x] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/illumina/parser/readers/CbclReader.java [hunk 1]
```diff
@@ -332,28 +332,31 @@ private void cacheTile(final int totalCycleCount, final TileData tileData, final
...
-        }
-        if (totalRead != tileData.uncompressedBlockSize) {
-            throw new PicardException(String.format("Error while decompressing from BCL file for cycle %d. Offending file on disk is %s",
-                    (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()));
+            if (totalRead != tileData.uncompressedBlockSize) {
+                throw new PicardException(String.format("Error while decompressing from BCL file for cycle %d. Offending file on disk is %s",
+                        (totalCycleCount + 1), this.streamFiles[totalCycleCount].getAbsolutePath()));
+            }
+        } else {
+            log.warn("Ignoring tile " + tileData.tileNum + " there are no PF reads.");
```
Comment (pshapiro4broad) [inline src/main/java/picard/illumina/parser/readers/CbclReader.java]: If this was in a method, you could do the negative case first, with a `return`, which IMO would be clearer.

**Comment by sooheelee** [comment]
Just to be clear @jacarey, before this fix the tool would error on no data tiles?

**Comment by jacarey** [comment]
@sooheelee No there was just no warning. We didn't realize you could have tiles defined without data. This is a valid state, but one that seemed relevant to anyone running this program so we output a warning.

**Comment by sooheelee** [comment]
Ok, thanks @jacarey. I will correct the updated release notes.

**Pull request merged**